soroban-spec/src/gen/ts/project_template/src/invoke.ts

```diff
@@ -0,0 +1,147 @@
+ */
+export async function invoke({ method, args = [], fee = 100 }: InvokeArgs): Promise<(TxResponse & { xdr: string }) | Simulation> {
+  const account = await getAccount()
+
+  const contract = new SorobanClient.Contract(CONTRACT_ID)
+
+  let tx = new SorobanClient.TransactionBuilder(account, {
+    fee: fee.toString(10),
+    networkPassphrase: NETWORK_PASSPHRASE,
+  })
+    .addOperation(contract.call(method, ...args))
+    .setTimeout(SorobanClient.TimeoutInfinite)
+    .build()
+
+  const simulated = await Server.simulateTransaction(tx)
+
+  const auths = simulated.results?.[0]?.auth
+
+  // is it possible for `auths` to be present but empty? Probably not, but let's be safe.
+  if (auths?.length > 0) {
+    if (auths.length > 1) {
+      throw new NotImplementedError("Multiple auths not yet supported")
+    }
+
+    const auth = SorobanClient.xdr.ContractAuth.fromXDR(auths[0], 'base64')
+
+    if (auth.addressWithNonce() !== undefined) {
+      throw new NotImplementedError(
+        `This transaction needs to be signed by ${auth.addressWithNonce()
+        }; how do we do that?`
+      )
+    }
+
+    const raw = await invokeRpc(tx, simulated);
+    return {
+      ...raw,
+      xdr: raw.resultXdr,
+    };
+  }
+
+  const { results } = await Server.simulateTransaction(tx)
+  if (!results || results[0] === undefined) {
+    if (simulated.error) {
+      throw new Error(simulated.error as unknown as string)
+    }
+    throw new Error(`Invalid response from simulateTransaction:\n{simulated}`)
+  }
+  return results[0]
+}
+
+
+async function invokeRpc(tx: Tx, simulation: SorobanClient.SorobanRpc.SimulateTransactionResponse): Promise<TxResponse> {
+  // Simulate the tx to discover the storage footprint, and update the
+  // tx to include it. If you already know the storage footprint you
+  // can use `addFootprint` to add it yourself, skipping this step.
+  tx = SorobanClient.assembleTransaction(tx, NETWORK_PASSPHRASE, simulation) as Tx
+
+  // sign with Freighter
+  const signed = await signTransaction(tx.toXDR(), {
+    networkPassphrase: NETWORK_PASSPHRASE,
+  })
+
+  // re-assemble with signed tx
+  tx = SorobanClient.TransactionBuilder.fromXDR(
+    signed,
+    NETWORK_PASSPHRASE
+  ) as Tx
+
+  const sendTransactionResponse = await Server.sendTransaction(tx);
+  let getTransactionResponse = await Server.getTransaction(sendTransactionResponse.hash);
+
+  const secondsToWait = 10
+  const waitUntil = new Date((Date.now() + secondsToWait * 1000)).valueOf()
+
+  let waitTime = 1000;
+  let exponentialFactor = 1.5
+
+  while ((Date.now() < waitUntil) && getTransactionResponse.status === "NOT_FOUND") {
+    // Wait a beat
+    await new Promise(resolve => setTimeout(resolve, waitTime))
+    /// Exponential backoff
+    waitTime = waitTime * exponentialFactor;
+    // See if the transaction is complete
+    getTransactionResponse = await Server.getTransaction(sendTransactionResponse.hash)
+  }
```

Comment:
This will actually wait about 11 seconds.. Because of the exponent, but whatever. 😂